src/kbmod/image_collection.py

```diff
@@ -0,0 +1,727 @@
+        """
+        # maybe timespan?
+        return self.data["mjd"][-1] - self.data["mjd"][0]
+
+    def toImageStack(self):
+        """Return an `~kbmod.search.image_stack` object for processing with
+        KBMOD.
+
+        Returns
+        -------
+        imageStack : `~kbmod.search.image_stack`
+            Image stack for processing with KBMOD.
+        """
+        # unpack the layred image list to flatten the array
+        # this is so stupidly costly because we have an internal array
+        # representation that doesn't interface with numpy via ndarray it makes
+        # a copy every time
+        layeredImages = [img for std in self._standardizers for img in std.toLayeredImage()]
+        return image_stack(layeredImages)
+
+    def _calc_suggested_angle(self, wcs, center_pixel=(1000, 2000), step=12):
+        """Projects an unit-vector parallel with the ecliptic onto the image
+        and calculates the angle of the projected unit-vector in the pixel
+        space.
+
+        Parameters
+        ----------
+        wcs : ``astropy.wcs.WCS``
+            World Coordinate System object.
+        center_pixel : tuple, array-like
+            Pixel coordinates of image center.
+        step : ``float`` or ``int``
+            Size of step, in arcseconds, used to find the pixel coordinates of
+                the second pixel in the image parallel to the ecliptic.
+
+        Returns
+        -------
+        suggested_angle : ``float``
+            Angle the projected unit-vector parallel to the ecliptic
+            closes with the image axes. Used to transform the specified
+            search angles, with respect to the ecliptic, to search angles
+            within the image.
+
+        Note
+        ----
+        It is not neccessary to calculate this angle for each image in an
+        image set if they have all been warped to a common WCS.
+
+        See Also
+        --------
+        run_search.do_gpu_search
+        """
+        # pick a starting pixel approximately near the center of the image
+        # convert it to ecliptic coordinates
+        start_pixel = np.array(center_pixel)
+        start_pixel_coord = SkyCoord.from_pixel(start_pixel[0], start_pixel[1], wcs)
+        start_ecliptic_coord = start_pixel_coord.geocentrictrueecliptic
+
+        # pick a guess pixel by moving parallel to the ecliptic
+        # convert it to pixel coordinates for the given WCS
+        guess_ecliptic_coord = SkyCoord(
+            start_ecliptic_coord.lon + step * u.arcsec,
+            start_ecliptic_coord.lat,
+            frame="geocentrictrueecliptic",
+        )
+        guess_pixel_coord = guess_ecliptic_coord.to_pixel(wcs)
+
+        # calculate the distance, in pixel coordinates, between the guess and
+        # the start pixel. Calculate the angle that represents in the image.
+        x_dist, y_dist = np.array(guess_pixel_coord) - start_pixel
+        return np.arctan2(y_dist, x_dist)
+
+    def run(self, config):
```

Comment:
Yes! This was more for demo and because this function drives requirements. 

Completely unrelated to this, I decided to go on a rant and write it down so I don't forget:

Looking at it, we can't really run processing because we can't avoid calling 

```python
suggested_angle = self._calc_suggested_angle(self.wcs[0], center_pixel)
```
This is where the assumptions that all the images are looking at the same point in the sky and that they're all the same sizes is coded. I think it's instructive to think about what the interface to the processing is. 

As is - processing basically doesn't require any state tracking, so a class wouldn't have any state keeping that the current `Results` don't already. So it doesn't feel like it should be a class. Really, it already is a class - the `KBMODSearch` or `search` in the code below.

So what the `run_search` would do is just unravel the config parameters and move the data from Python to C++. 
And either of those are really functions of things that already exist. So I do wonder whether or not this whole thing can be made to just be a `run(ImageCollection, config)` convenience function in the sense that it returns the filtered results and not the `KBMODSearch` object and that we should improve the `KBMODSearch` interface.

I think the hard thing about doing this refactoring is developing a meaningful, purposeful plan forward where we don't just "add a feature to support mapping multiple instruments to KBMOD abstraction", but holistically develop a cohesive interface that brings the code together and makes it act like a package and not a collection of functionalities. 

This includes reasoning about interfaces between the data abstraction code in C++, Python, their interaction with the data processing code (both in C++ and in filtering in Python) and the end user interface in the form of Results. I'm not for accepting this abstraction into the code if all we want is a mapper between different data sources and KBMOD processing. That can also be achieved by making https://github.com/dirac-institute/kbmod/blob/94cb6d75c139de8fbd03082e3b533e50cccefc0b/src/kbmod/image_info.py#L28
an abstract method and then providing a bunch of default mapper functions like `decam_mapper(file_path) --> dict` and then saying `ImageInfo(mapper=decam_mapper)` or `ImageInfo(callable=my_mapper)` and keep no states or complexity like we do here.

Reply:
I strongly agree on needing to purposefully plan a cohesive interface.